src/components/button/button.tsx

```diff
@@ -332,6 +334,14 @@ export class Button
     this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
   }
 
+  /** inital full length slotted text before the truncation */
+  private fullLengthSlottedText: string;
```

Comment:
I think this should be `@State()` so that it triggers a re-render when this value changes.

Reply:
Maybe we can think of a better name like just `fullText` or `tooltipText`?